play/src/components/finalresults/Leaderboard.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useRef, useEffect, useState } from 'react';
+import {
+  GameSessionState,
+  ITeam,
+  isNullOrUndefined,
+} from '@righton/networking';
+import { v4 as uuidv4 } from 'uuid';
+import { Grid } from '@mui/material';
+import HeaderContent from '../HeaderContent';
+import FooterContent from '../FooterContent';
+import StackContainerStyled from '../../lib/styledcomponents/layout/StackContainerStyled';
+import HeaderStackContainerStyled from '../../lib/styledcomponents/layout/HeaderStackContainerStyled';
+import BodyStackContainerStyled from '../../lib/styledcomponents/layout/BodyStackContainerStyled';
+import BodyBoxUpperStyled from '../../lib/styledcomponents/layout/BodyBoxUpperStyled';
+import BodyBoxLowerStyled from '../../lib/styledcomponents/layout/BodyBoxLowerStyled';
+import { BodyContentAreaLeaderboardStyled } from '../../lib/styledcomponents/layout/BodyContentAreaStyled';
+import FooterStackContainerStyled from '../../lib/styledcomponents/layout/FooterStackContainerStyled';
+import 'swiper/css';
+import 'swiper/css/pagination';
+import LeaderboardSelector from '../LeaderboardSelector';
+
+interface LeaderboardProps {
+  teams?: ITeam[];
+  currentState: GameSessionState;
+  teamAvatar: number;
+  teamId: string;
+  score: number;
+  isSmallDevice: boolean;
+}
+
+export default function Leaderboard({
+  teams,
+  currentState,
+  teamAvatar,
+  teamId,
+  score,
+  isSmallDevice,
+}: LeaderboardProps) {
+  const currentTeam = teams?.find((team) => team.id === teamId);
+  const teamSorter = (inputTeams: ITeam[]) => {
+    inputTeams.sort((a, b) => {
+      if (a.score < b.score) {
+        return 1;
+      }
+      if (a.score > b.score) {
+        return -1;
+      }
+      return 0;
+    });
+    return teams;
+  };
+
+  let sortedTeams;
+  if (!isNullOrUndefined(teams)) {
+    sortedTeams = teamSorter(teams);
+  }
+
+  // this gets the height of the container ref and then adjusts the height of the subcontainer for the leaderboard so there isn't any partial overflow
+  const containerRef = useRef<HTMLDivElement>(null); // ref req'd for height of container
+  const itemRef = useRef<HTMLDivElement>(null); // ref req'd for height of item
+  const [subContainerHeight, setSubContainerHeight] = useState<number>(0); // height of subcontainer
+
+  useEffect(() => {
+    if (containerRef.current && itemRef.current) {
+      // check if the container element has been loaded yet
+      const containerHeight = containerRef.current.clientHeight; // get height of container
+      const itemHeight = itemRef.current.clientHeight; // get height of item
+      setSubContainerHeight(containerHeight - (containerHeight % itemHeight)); // adjust height of subcontainer to be a multiple of the item height
```

Comment:
nit: you usually wanna put the comment on a separate line above the code. Helps with having shorter lines and easier to read

Reply:
Updated. I know you've mentioned this before so I'll try to clean up the commenting as I go to follow this. The only exception for this block specifically is ln 70 (the dependency array). It seemed odd to split the end of the useEffect with a comment, given the comment is short. let me know though if this is an issue